### ISSUE TITLE
[FEATURE] Add ML-based imputation

### ISSUE THREAD

**Issue by eschibli** [unit8co/darts#2870]
**Is your feature request related to a current problem? Please describe.**
Not a problem, though this functionality was [recently asked about](https://github.com/unit8co/darts/issues/2867)

**Describe proposed solution**
It would be rather simple to include one or more forecaster-based imputers, which could offer significantly superior filling of missing values than simple interpolation methods. Perhaps most simply we could include an XGBoost- or LightGBM-based regression imputer, as these models can handle NaN features natively without trouble. This could be accomplished by treating all observations as known covariates, allowing both lookback and lookahead. Then the missing values in the target series could be filled with indicator values, and unweighted to zero to prevent fitting on them. 

Something like 
```


# 'forecaster' is an XGBoost or LightGBM forecaster 
# with lags=None, lags_future_covariates = [-lookback, ..., -1, 1, ..., lookahead]
class BoostedTreesImputer:
   def __init__(self, forecaster, lookback, lookahead, lags=None)
...
   def impute(self, ts):
      dummy_filled_ts = fill_nans(ts, -1E6)
      ts_nan_padded = pad_ts(ts, lookback, lookahead)

      self.forecaster.fit(dummy_filled_ts, future_covariates=ts_nan_padded)

      back_predicted_ts = self.forecaster.historical_forecasts(..., forecast_horizon=1, retrain=False) 
      
      imputed_ts = fill_nans_only(ts, back_predicted_ts)
      return imputed_ts
```

This would also not be too difficult to extend to an iterative imputer like [MissForest](https://arxiv.org/abs/1105.0828) or the scikit-learn [Iterative Imputer](https://scikit-learn.org/stable/modules/generated/sklearn.impute.IterativeImputer.html) that could use any traditional interpolation method as a baseline to iterate on. This could work with arbitrary Forecasters.

**Limitations of proposed solutions**
The native nan-aware models in particular could give unexpectedly degraded performance in cases where the missing values are not uniformly distributed. 

There doesn't seem to be a trivial way to allow the use of contemporaneous cross-component information without allowing for catastrophic overfitting on targets 

A traditional autoencoder-based solution that encodes and decodes a fixed-with chunk with heavy input dropout in training would probably be more optimal than any of the forecasting-oriented neural network models, but doesn't seem simple to implement in the Darts framework to me. 

**Describe potential alternatives**
These workflows aren't difficult to achieve without a dedicated class, but I do think offering a validated a debugged solution would be valuable.

**Comment by dennisbader** [comment]
Hi @eschibli and thanks for the feature request! (sorry for the late response, I was on holidays)

This is a really interesting idea - using forecasting models for imputation could definitely be more sophisticated than simple interpolation.
However, I think that at the moment we're not planning to add this to Darts. The main reason is that Darts is designed to provide robust, well-tested building blocks rather than highly specialized solutions. The proposed imputer would essentially be a wrapper around existing forecasting capabilities that users can already access directly.

You can:
- Use any of our forecasting models (XGBoost, LightGBM, etc.) with custom lag configs
- Leverage the existing MissingValuesFiller for initial imputation
- Create custom transformers using our base classes
- Chain everything together with the Pipeline class

This approach gives you maximum flexibility while keeping Darts focused on its core strengths.
